module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Logic/RenewalsHelperTest.php

```diff
@@ -0,0 +1,433 @@
+<?php
+
+/**
+ * VuFind Helper - Renewals Support Methods
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2010.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Sambhav Pokharel <sambhavpokharel@gmail.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFindTest\ILS\Logic;
+
+use Laminas\Mvc\Plugin\FlashMessenger\FlashMessenger;
+use Laminas\Stdlib\Parameters;
+use PHPUnit\Framework\TestCase;
+use VuFind\ILS\Connection;
+use VuFind\ILS\Logic\RenewalsHelper;
+use VuFind\Validator\CsrfInterface;
+
+use function in_array;
+use function is_array;
+
+/**
+ * Unit tests for the RenewalsHelper
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Sambhav Pokharel <sambhavpokharel@gmail.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class RenewalsHelperTest extends TestCase
+{
+    /**
+     * Test adding renew details when using an external link.
+     *
+     * @return void
+     */
+    public function testAddRenewDetailsWithLink(): void
+    {
+        $catalog = $this->getMockBuilder(Connection::class)
+            ->disableOriginalConstructor()
+            ->onlyMethods(['__call'])
+            ->getMock();
+
+        $ilsDetails = ['id' => 'item123', 'title' => 'Test Item'];
+        $renewStatus = ['function' => 'renewMyItemsLink'];
+        $expectedLink = '/MyResearch/renew/item123';
+
+        $catalog->expects($this->once())
+            ->method('__call')
+            ->with(
+                $this->equalTo('renewMyItemsLink'),
+                $this->equalTo([$ilsDetails])
+            )
```

Comment:
I think you may be able to simplify this to:

```suggestion
            ->with('renewMyItemsLink', [$ilsDetails])
```

(I may be misremembering, but I think one of the Rector PHPUnit optimizations that we may be introducing in the future was to simplify `$this->equalTo` assertions where they were not strictly necessary to reduce verbosity).